src/java/org/apache/cassandra/auth/IAuthenticator.java

```diff
@@ -179,5 +194,50 @@ default boolean shouldSendAuthenticateMessage()
         {
             return true;
         }
+
+        /**
+         * @return The assumed mode of authentication attempted using this negotiator, this will usually be some value
+         * of {@link AuthenticationMode#getDisplayName()} unless an implementor provides their own custom authentication
+         * scheme.
+         */
+        default String getAuthenticationMode()
+        {
+            return AuthenticationMode.UNAUTHENTICATED.getDisplayName();
+        }
+    }
+
+    /**
+     * Known modes of authentication supported by Cassandra's provided {@link IAuthenticator} implementations.
+     */
+    enum AuthenticationMode
+    {
+        /**
+         * User was not authenticated in any particular way.
+         */
+        UNAUTHENTICATED,
+        /**
+         * User authenticated using a password.
+         */
+        PASSWORD,
+        /**
+         * User authenticated using a trusted identity in their client certificate.
+         */
+        MTLS;
+
+        private final String displayName;
+
+        AuthenticationMode()
+        {
+            this.displayName = StringUtils.capitalize(name().toLowerCase());
+        }
+
+        /**
+         * @return How the mode should be displayed to users. Formally it is the enum name with the first letter
```

Comment:
@tolbertam aha, this is interesting ... why have you chose such a naming convention? Why not just `AuthenticationMode.name()` ?

Reply:
I thought `Password` would be more friendly than `PASSWORD` and `name()` returns the all caps representation of the enum (and is not overridable)